drivers/pinctrl/pinctrl_stm32.c

```diff
@@ -0,0 +1,588 @@
+			*func = LL_GPIO_GetAFPin_0_7(ll_gpio_port,
+						     ll_gpio_pin);
+		} else {
+			*func = LL_GPIO_GetAFPin_8_15(ll_gpio_port,
+						      ll_gpio_pin);
+		}
+		*func += PINCTRL_STM32_FUNCTION_ALT_0;
+		break;
+	case LL_GPIO_MODE_OUTPUT:
+		*func = PINCTRL_STM32_FUNCTION_OUTPUT;
+		break;
+	case LL_GPIO_MODE_INPUT:
+		*func = PINCTRL_STM32_FUNCTION_INPUT;
+		break;
+	case LL_GPIO_MODE_ANALOG:
+		*func = PINCTRL_STM32_FUNCTION_ANALOG;
+		break;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Set muxing function at pin
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param pin Pin
+ * @param func Muxing function
+ * @retval 0 on success
+ * @retval -ENOTSUP if requested pin is not available on this controller
+ * @retval -EINVAL if requested function is unknown
+ */
+static int pinctrl_stm32_mux_set(struct device *dev, u16_t pin, u16_t func)
+{
+	if (pin >= PINCTRL_STM32_PIN_COUNT) {
+		return -ENOTSUP;
+	}
+
+	const struct pinctrl_lib_config * config = 
+					pinctrl_lib_device_config_info(dev);
+
+	if (func < config->function_count) {
+		/* func denotes a device 
+		 * get hardware pinmux control for pin 
+		 * from a pinctrl for the device function. */
+		for (int i = 0; i < config->pinctrl_count; i++) {
+			u16_t pinctrl_state = config->pinctrl_data[i].state;
+			u16_t pinctrl_func = config->pinctrl_state_data[
+						pinctrl_state].function;
+			u16_t pinctrl_pin = config->pinctrl_data[i].pin;
+			if ((pinctrl_func == func) && (pinctrl_pin == pin)) {
+				func = config->pinctrl_data[i].mux;
+				break;
+			}
+		}
+	}
+	if (func < PINCTRL_FUNCTION_PINMUX_BASE) {
+		return -EINVAL;
+	}
+	/* now func denotes a hardware pinmux control */
+	GPIO_TypeDef *ll_gpio_port = pinctrl_stm32_ll_gpio_port(pin);
+	u32_t ll_gpio_pin = pinctrl_stm32_ll_gpio_pin(pin);
+
+	if (pinctrl_stm32_ll_gpio_alt_func(func) <= LL_GPIO_AF_7) {
+		/* valid alternate function */
+		if (pinctrl_stm32_ll_gpio_pin(pin) <= LL_GPIO_PIN_7) {
+			LL_GPIO_SetAFPin_0_7(ll_gpio_port, ll_gpio_pin,
+					pinctrl_stm32_ll_gpio_alt_func(func));
+		} else {
+			LL_GPIO_SetAFPin_8_15(ll_gpio_port, ll_gpio_pin,
+					pinctrl_stm32_ll_gpio_alt_func(func));
+		}
+	}
+	LL_GPIO_SetPinMode(ll_gpio_port, ll_gpio_pin,
+			   pinctrl_stm32_ll_gpio_mode(func));
+	return 0;
+}
+
+/**
+ * @brief Initialise pin controller.
+ * 
+ * Enable all GPIOS controlled by the pin controller (currently all GPIOs)
+ * to be clocked by the associated peripherial bus clock.
+ * 
+ * @param dev Pointer to the device structure for the driver instance.
+ */
+static int pinctrl_stm32_device_init(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	/* Enable all GPIOS to be clocked by the associated peripherial bus clock */
+	pinctrl_stm32_data.clk = device_get_binding(STM32_CLOCK_CONTROL_NAME);
+	pinctrl_stm32_data.pclken_gpio.bus = 
+		CHAOS_PP_CAT(STM32_CLOCK_BUS_, PINCTRL_STM32_LL_BUS_GPIO);
+	for(int port_idx = 0; port_idx < PINCTRL_STM32_GPIO_COUNT;
+	    port_idx++) {
```

Comment:
I'm not sure this for() line passes checkpatch !

Reply:
Changed. Thank you.